File: Data/Entities/Meal.cs
namespace FoodDiary.Data.Entities
{
    public class Meal
    {
        public int Id { get; set; }
        public User User { get; set; }
        public int UserId { get; set; }
        public DateTime MealTime { get; set; }

        public ICollection<MealItem> MealItems { get; set; }
    }
}

File: client/src/app/shared/User/Meal.ts
import { MealItem } from "./MealItem";
import { MealValue } from "./MealValue";

export class Meal {

    id: number = 0;
    mealTime: Date = new Date();
    mealItems: MealItem[] = [];       
    mealValue: MealValue = new MealValue();
}